meilisearch/src/search.rs

```diff
@@ -1045,6 +1064,24 @@ fn make_hits(
         let mut document =
             permissive_json_pointer::select_values(&displayed_document, attributes_to_retrieve);
 
+        if retrieve_vectors {
+            let mut vectors = serde_json::Map::new();
+            for (name, mut vector) in index.embeddings(rtxn, id)? {
+                let user_defined = embedding_configs
+                    .iter()
+                    .find(|conf| conf.name == name)
+                    .is_some_and(|conf| conf.user_defined.contains(id));
+                let mut embedding = serde_json::Map::new();
+                embedding.insert("userDefined".to_string(), user_defined.into());
+                match vector.as_mut_slice() {
+                    [one] => embedding.insert("embedding".to_string(), std::mem::take(one).into()),
```

Comment:
I think it is `embeddings` here (with an s).

Should we maybe use an `ExplicitVector` object that we serialize like in the dumps?

Reply:
Yes you’re absolutely right, that was buggy good catch